x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/lucene/ValuesSourceReaderOperator.java

```diff
@@ -681,7 +681,12 @@ public BlockLoader.Builder nulls(int expectedCount) {
 
         @Override
         public Block constantNulls() {
-            if (nullBlock == null) {
+            // Avoid creating a new null block if we already have one.
+            // However, downstream operators take ownership of the null block we hand to them and may close it, forcing us to create a new
+            // one.
+            // This could be avoided altogether if this factory itself kept a reference to the null block, but we'd have to also ensure
+            // to release this block once the factory is not used anymore.
```

Comment:
Right, the factory itself would have to be `Releasable`. FWIW, that's not really a big deal. I think I'd prefer it to this fix to be honest. I'm kind of ok with whichever though.

Reply:
But if you are trying to get this merged quickly I'd be fine with this solution - especially if you are willing to try the other in a followup.

Reply:
I also prefer making this `Releasable`. It's easy to do, I'll push a commit with the change.

Wrong implementations of `.close()` or forgetting to incref when handing out blocks actually nicely show up as test failures already in the existing tests, so I think this is guarding better against regressions.